From 7ef23635252e4707f1e26a8dcada84e79195bcb9 Mon Sep 17 00:00:00 2001
From: Karthikeyan Kathirvel <quic_kathirve@quicinc.com>
Date: Tue, 18 Apr 2023 18:08:25 +0530
Subject: [PATCH] wpa_supplicant: mesh: Allow ACL command

As part of extended AP changes ACL command APIs has been changed.

Allow ACL commands for mesh mode.

Fixes: 9828aba16eee ("Support ACL operations in wpa_supplicant AP mode")

Signed-off-by: Karthikeyan Kathirvel <quic_kathirve@quicinc.com>
---
 wpa_supplicant/ap.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

--- a/wpa_supplicant/ap.c
+++ b/wpa_supplicant/ap.c
@@ -1684,6 +1684,8 @@ int ap_ctrl_iface_acl_add_mac(struct wpa
 
 	if (wpa_s->ap_iface)
 		hapd = wpa_s->ap_iface->bss[0];
+	else if (wpa_s->ifmsh)
+		hapd = wpa_s->ifmsh->bss[0];
 	else
 		return -1;
 
@@ -1710,6 +1712,8 @@ int ap_ctrl_iface_acl_del_mac(struct wpa
 
 	if (wpa_s->ap_iface)
 		hapd = wpa_s->ap_iface->bss[0];
+	else if (wpa_s->ifmsh)
+		hapd = wpa_s->ifmsh->bss[0];
 	else
 		return -1;
 
@@ -1736,6 +1740,8 @@ int ap_ctrl_iface_acl_show_mac(struct wp
 
 	if (wpa_s->ap_iface)
 		hapd = wpa_s->ap_iface->bss[0];
+	else if (wpa_s->ifmsh)
+		hapd = wpa_s->ifmsh->bss[0];
 	else
 		return -1;
 
@@ -1759,6 +1765,8 @@ void ap_ctrl_iface_acl_clear_list(struct
 
 	if (wpa_s->ap_iface)
 		hapd = wpa_s->ap_iface->bss[0];
+	else if (wpa_s->ifmsh)
+		hapd = wpa_s->ifmsh->bss[0];
 	else
 		return;
 
@@ -1779,6 +1787,8 @@ int ap_ctrl_iface_disassoc_deny_mac(stru
 
 	if (wpa_s->ap_iface)
 		hapd = wpa_s->ap_iface->bss[0];
+	else if (wpa_s->ifmsh)
+		hapd = wpa_s->ifmsh->bss[0];
 	else
 		return -1;
 
@@ -1792,6 +1802,8 @@ int ap_ctrl_iface_disassoc_accept_mac(st
 
 	if (wpa_s->ap_iface)
 		hapd = wpa_s->ap_iface->bss[0];
+	else if (wpa_s->ifmsh)
+		hapd = wpa_s->ifmsh->bss[0];
 	else
 		return -1;
 
@@ -1805,6 +1817,8 @@ int ap_ctrl_iface_set_acl(struct wpa_sup
 
 	if (wpa_s->ap_iface)
 		hapd = wpa_s->ap_iface->bss[0];
+	else if (wpa_s->ifmsh)
+		hapd = wpa_s->ifmsh->bss[0];
 	else
 		return -1;
 
